### PR TITLE
Wait for async completion in integration tests

### PR DESCRIPTION
**Pull request by sharwell**
A test failure was observed where `BasicIntellisense.TypeAVariableDeclaration` failed on this line:

https://github.com/dotnet/roslyn/blob/5d2310a380ce7d00284cd821d5650230370e583a/src/VisualStudio/IntegrationTest/IntegrationTests/VisualBasic/BasicIntelliSense.cs#L152

The failure occurred in an async completion scenario. Since <kbd>Space</kbd> is a completion trigger, there is a period of time where Roslyn tells the IDE that completion *should* show, and later when no completion items are found the session is dismissed. We need to wait for this computation to complete before making assertions about its outcome.

### CODE REVIEW

**Comment by AmadeusW** [comment]
Here's how we do it in Editor.
Broker's CompletionTriggered is fired synchronously. This is the earliest and the most reliable way to get a hold of the completion session. Use this opportunity to hook up to its events: `ItemsUpdated`, `ItemCommitted`, `Dismissed`

```
            var sessionComputedEvent = new AutoResetEvent(false);
            Broker.CompletionTriggered += (s, triggerArgs) =>
            {
                triggerArgs.CompletionSession.ItemsUpdated += (o, e) => sessionComputedEvent.Set();
            };
```

and then, in the test:
```
await sessionComputedEvent.ToTask(MaxTestDelay);
```

**Comment by sharwell** [comment]
@AmadeusW can you take another look? I updated it to map the approach you described to the operation tracker used elsewhere in Roslyn.

File: src/VisualStudio/IntegrationTest/TestSetup/IntegrationTestServiceCommands.cs [hunk 1]
```diff
@@ -130,6 +135,10 @@ private void StopServiceCallback(object sender, EventArgs e)
                 _marshalledService = null;
                 _service = null;
 
+                var componentModel = ServiceProvider.GetService<SComponentModel, IComponentModel>();
+                var asyncCompletionTracker = componentModel.GetService<AsyncCompletionTracker>();
+                asyncCompletionTracker.StartListening();
```
Comment (jasonmalinowski) [inline src/VisualStudio/IntegrationTest/TestSetup/IntegrationTestServiceCommands.cs]:
```suggestion
                asyncCompletionTracker.StopListening();
```

Comment (sharwell) [reply src/VisualStudio/IntegrationTest/TestSetup/IntegrationTestServiceCommands.cs]: fixed!

**Pull request merged**